cmake/thirdparty/get_spdlog.cmake

```diff
@@ -17,6 +17,14 @@ function(find_and_configure_spdlog)
 
   include(${rapids-cmake-dir}/cpm/spdlog.cmake)
   rapids_cpm_spdlog(
+    # The conda package for fmt is hard-coded to assume that we use a preexisting fmt library. This
```

Comment:
Is it viable to remove conda package dependencies and always get spdlog/fmt as header-only deps via CPM?

Reply:
> Is it viable to remove conda package dependencies

No. The problem is that the logger is compiled not by rmm itself but by packages that build against rmm, and we cannot guarantee that spdlog or fmt will never exist in the environment in which that build is taking place. For example, if cuml has some other dependency that requires spdlog, then during cuml's conda build spdlog will be in the environment when its copy of the rmm logger is built. You can think of how the new build works as similar to what we do with gtest/gmock, with the additional complexity that the build is done by the user of rmm rather than rmm itself.

> always get spdlog/fmt as header-only deps via CPM

Yes, with caveats. This is what I was alluding to in [my message above](https://github.com/rapidsai/rmm/pull/1739#issuecomment-2502792166) where I said

> We could make the first somewhat more palatable by having the download be controlled by some flag that is set to on by default (since that's what we'll want for everything in RAPIDS). I will prototype that after this PR is merged.

The big caveat is that we do not want to require this all the time, otherwise it will make offline builds of any package built against rmm impossible because you will have CPMAddPackage calls embedded in a dependency (the only workarounds would be to manually modify CPM-cloned code inside the build directory). We need to write the CMake so as to make this behavior possible to opt out of.